zenoh-kotlin/src/commonMain/kotlin/io/zenoh/Logger.kt

```diff
@@ -15,13 +15,35 @@
 package io.zenoh
 
 /** Logger class to redirect the Rust logs from Zenoh to the kotlin environment. */
-class Logger {
+internal class Logger {
 
     companion object {
+
+        internal const val LOG_ENV: String = "RUST_LOG"
+
+        fun start(logLevel: LogLevel) = runCatching {
+            startLogsViaJNI(logLevel.description)
+        }
+
         /**
          * Redirects the rust logs either to logcat for Android systems or to the standard output (for non-android
-         * systems). @param logLevel must be either "info", "debug", "warn", "trace" or "error".
+         * systems).
+         *
+         * @param logLevel must be either "info", "debug", "warn", "trace" or "error".
          */
-        external fun start(logLevel: String)
+        @Throws(Exception::class)
+        private external fun startLogsViaJNI(logLevel: String)
+    }
+}
+
+enum class LogLevel(internal val description: String) {
```

Comment:
loglevel may be more complex than that, see https://docs.rs/env_logger/latest/env_logger/#enabling-logging
Just pass the raw string to Rust and let it cook?